src/transformers/modeling_utils.py

```diff
@@ -2090,9 +2161,19 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             # Make sure tied weights are tied before creating the device map.
             model.tie_weights()
             device_map = infer_auto_device_map(
-                model, no_split_module_classes=no_split_modules, dtype=torch_dtype, max_memory=max_memory
+                model,
+                no_split_module_classes=no_split_modules,
+                dtype=torch_dtype if not load_in_8bit else torch.int8,
+                max_memory=max_memory,
             )
 
+            if load_in_8bit:
+                # The LM head can stay on disk / CPU
+                device_map_without_lm_head = {key: device_map[key] for key in device_map.keys() if key != "lm_head"}
```

Comment:
This is a bit too adhoc here (though I understand this is a bit anecdotic). Some models may call it by a different name. You should rely on the methods of `PreTrainedModel` that give you the output embeddings here, probably.

Reply:
I have the same comment as below, I am proposing a solution in 279b8c45bcf3893e1b4695d32ff2e1083c6bda9d which should be clear for that